internal/api/v1/application/create.go

```diff
@@ -131,3 +141,64 @@ func (hc Controller) Create(c *gin.Context) apierror.APIErrors {
 	response.Created(c)
 	return nil
 }
+
+func validateRoutes(ctx context.Context, cluster *kubernetes.Cluster, appName, namespace string, desiredRoutes []string) apierror.APIErrors {
+	log := requestctx.Logger(ctx)
+
+	desiredRoutesMap := map[string]struct{}{}
+	for _, desiredRoute := range desiredRoutes {
+		desiredRoutesMap[desiredRoute] = struct{}{}
+	}
+
+	ingressList, err := cluster.Kubectl.NetworkingV1().Ingresses("").List(ctx, metav1.ListOptions{})
```

Comment:
Maybe just set the app name to `unknown, external`. That should allow us to keep the general structure of the code, and messages.

Reply:
Actually it will still return an error, because of the clashing route, it's just going to skip the "ownership" check. I would not set the name to a "reserved" word, because there would be the possibility of strange behaviors in case someone decide to deploy an app with a weird name, such as `external`.

Having a fresh look at the code I've left a couple of logs that I've used to debug. And also that checking the ownership of the app without distinguish the app name from the namespace it's a source of error. I.e.: what if an app is called `foobar` in the namespace `baz` and there is also a `foo` app in the `barbaz` namespace? It will result ok in the check, while it's not.

Reply:
> I would not set the name to a "reserved" word, because ...

Right. In these kind of situations I distinguish between `app name`, and `displayed app name`.
The first for functionality which needs the name, the other for use in messages.
That way there is no reserved word around which can bamboozle checks and conditions.